ArcadiaParties.API/Controllers/UsersController.cs

```diff
@@ -0,0 +1,55 @@
+﻿using ArcadiaParties.CQRS.Queries;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+using System.Security.Principal;
+using System.Threading.Tasks;
+
+namespace ArcadiaParties.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class UsersController : ControllerBase
+    {        
+        private readonly IMediator _mediator;
+
+        public UsersController(IMediator mediator)
+        {
+            _mediator = mediator;
+        }
+
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [SwaggerOperation(
+             Summary = "Returns all users"
+        )]
+        [Authorize(Roles = "Administrator")]
+        [HttpGet]
+        [Route("GetUsers")]
+        public async Task<IActionResult> GetUsers()
+        {
+            var query = new GetAllUsersQuery();
+            var users = _mediator.Send(query);
+
+            return Ok(users);
+        }
+
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [SwaggerOperation(
+             Summary = "Returns current authenticated Windows user"
+        )]
+        [HttpGet]
+        [Route("GetCurrentUser")]
+        public async Task<IActionResult> GetCurrentUser()
+        {
+            IPrincipal principal = HttpContext.User;
```

Comment:
No. Why do you need `HttpContext`, if you didn't use it before?

Reply:
Sure. I've removed it.